Code/BuildSystem/CMake/CMakeUtils/ezUtilsTarget.cmake

```diff
@@ -155,4 +155,20 @@ macro(ez_create_target TYPE TARGET_NAME)
 		set_property(GLOBAL APPEND PROPERTY "EXPORT_PROJECTS" ${TARGET_NAME})
 	endif()
 
+	if(EZ_SUPPORT_TRACY)
+		# Tracy has to be linked to all possible targets since it is a multi dll project.
+		if(TARGET TracyClient)
+			target_compile_definitions(${TARGET_NAME} PRIVATE
+				TRACY_EXPORTS
+				TRACY_ENABLE
+				TRACY_NO_INVARIANT_CHECK
+				TRACY_NO_FRAME_IMAGE
+			)
+
+			if(NOT ${TARGET_NAME} STREQUAL "TracyClient" OR "Tracy::TracyClient")
+				target_link_libraries(${TARGET_NAME} PRIVATE TracyClient)
+			endif()
+		endif()
+	endif()
+	
```

Comment:
This shouldn't be here.

The compile definitions should be set on the Tracy library as PUBLIC, then they will automatically propagate to linked targets.

ezFoundation should be the only library that links against tracy, and it should link PUBLIC against it, so that the PUBLIC definitions of tracy propagate to its users.

The EZ_SUPPORT_TRACY option should be set up in an init.cmake file in the tracy lib.

Reply:
Resolved in latest commit.